pre_commit_hooks/sort_cspell_json.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 Tier IV, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+from pathlib import Path
+
+
+# The key contained in the json.
+# The default setting is {unique: False, sort: False, key_order: None}
+KEYS = {
+    "ignorePaths": {"unique": True, "sort": True, "key_order": None},
+    "words": {"unique": True, "sort": True, "key_order": str.lower},
+}
+
+
+def unique_json(data, key):
+    if key in data:
+        data[key] = list(set(data[key]))
+    return data
+
+
+def sort_json(data, key, key_order=None):
+    if key in data:
+        data[key] = sorted(data[key], key=key_order)
+    return data
+
+
+def parse(data):
+    for key in KEYS:
+        if KEYS[key]["unique"]:
+            data = unique_json(data, key)
+        if KEYS[key]["sort"]:
+            data = sort_json(data, key, key_order=KEYS[key]["key_order"])
```

Comment:
Could you briefly explain this logic?

Reply:
I use the following dict.
There are three option for content: unique, sort, and key_order.
If unique is true, unique_json is performed on the corresponding key value.
If sort is true, sort_json is performed on the corresponding key value. key_order allows you to specify the order during sorting.
```
KEYS = {
    "ignorePaths": {"unique": True, "sort": True, "key_order": None},
    "words": {"unique": True, "sort": True, "key_order": str.lower},
}
```

Reply:
I've applied my suggestions. How about this? d6be47b1c88d5e5c11a71af416a1cd0641006223
I'm sorry for my drastic changes. :bow:

Reply:
* The results are different before and after your script change. Perhaps you changed the answer file for that.
* Why did you make a major change after the review was over? I do not understand why.

Reply:
> The results are different before and after your script change. Perhaps you changed the answer file for that.

Yes.

> Why did you make a major change after the review was over? I do not understand why.

What do you mean? :thinking: 
I think the review isn't over. It's my review and improvement.
If you want to split it into another PR, I can do it.

Reply:
Also, are the results consistent with the following commands in README?
https://github.com/tier4/autoware-spell-check-dict/blob/main/README.md?plain=1#L9

Reply:
> Also, are the results consistent with the following commands in README?

Yes, I think so.
But other keys are sorted as well.

Reply:
> > The results are different before and after your script change. Perhaps you changed the answer file for that.
> 
> Yes.
> 
> > Why did you make a major change after the review was over? I do not understand why.
> 
> What do you mean? thinking I think the review isn't over. It's my review and improvement. If you want to split it into another PR, I can do it.

I have made modifications according to your review. But putting in destructive changes after that review is problematic for the review process.
If you make any major changes in the future, please tell me before the review.
For the script, there is nothing wrong with the change, so I approve it.

Reply:
I'm sorry for my lack of explanation. :bow: